tests/ReferenceTest.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+use const DIRECTORY_SEPARATOR;
+
+class ReferenceTest extends TestCase
+{
+    use InvalidCodeAnalysisTestTrait;
+    use ValidCodeAnalysisTestTrait;
+
+    /**
+     * @return iterable<string,array{string,assertions?:array<string,string>,error_levels?:string[]}>
+     */
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'updateReferencedTypes' => [
+                '<?php
```

Comment:
can you test what happens if you change the value of $a after creating the reference in $b?

Reply:
Hmm, I thought I tested that sufficiently elsewhere, but now I'm not so sure.